### PR TITLE
Fix tree shaking

### PR DESCRIPTION
**Pull request by penx**
Export an es module rather than an object.

This is what radix primitives currently does. 

This ensures unused components are not included in a bundled application (tree shaking). 

Also see:

- https://twitter.com/penx/status/1590084633655316480
- https://github.com/penx/cra-tree-shaking-tests

### CODE REVIEW

**Comment by vladmoroz** [comment]
This is great, I hope we'll have a close look and test it thoroughly within a few weeks

**Comment by thisislvca** [comment]
This would be a really nice to have for sure!

**Comment by penx** [comment]
@benoitgrelard @vladmoroz do you want this in 2.1?

happy to do the work for rebase when you’re ready for it

**Comment by penx** [comment]
rebased it anyway 😄

File: packages/radix-ui-themes/src/components/alert-dialog.parts.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export { AlertDialogRoot as Root } from './alert-dialog';
```
[the remark this answers is not in the document]
Comment (penx) [reply packages/radix-ui-themes/src/components/alert-dialog.parts.ts:1]: I'm following what is already being done with the .props.ts files in this directory, but happy to update either way if maintainers have a preference

**Comment by vladmoroz** [comment]
Hey @penx, is the intention here to make sure tree-shaking works for individual parts when imported with dot notation?

**Comment by penx** [comment]
> Hey @penx, is the intention here to make sure tree-shaking works for individual parts when imported with dot notation?

yes that's right

**Review by vladmoroz — APPROVED** [review]
Thanks for the help @penx, merging now

**Comment by devfixit** [comment]
@penx @vladmoroz is there a way to tree shake css as well cuz i'm reexporting Flex component from radix-ui themes and have included the styles but when i build, entire css is getting generated.

```import '@radix-ui/themes/styles.css';
export { Flex } from "@radix-ui/themes"```

this is what flex.tsx looks like. on a side note i'm using shadcn-ui and it has lots of deps on radix-ui.
Thanks

**Comment by vladmoroz** [comment]
Could you tell more about your use-case? @devfixit

CSS tree shaking isn't really a thing, at least not in the same sense as relative to how JS tree shaking works.
In theory something like https://purgecss.com could work, but we haven't tested it. Might be non-trivial to set it up.

We were thinking about exporting a CSS file that would cover just the layout components and their props (Box, Grid, Flex, Section, Container). Would that be helpful for what you are trying to achieve?

**Comment by penx** [comment]
I'd be interested to see what the bundle size gains would be for people using ~50% of radix themes components, and if there was any significant improvement to browser parsing time. We're using a lot of radix themes at this point, and I think it's coming in at 76kb gzipped, so expect it's not too significant for our use case. 

I had been wondering about making a cli util that builds the radix themes css with postcss (with the radix themes internal postcss config) with certain components removed, but it would rely on using radix themes internals so could break in patch updates so to be honest I am not convinced development and maintenance time would be worth it.

**Comment by penx** [comment]
fwiw I don't think the impact of the javascript tree shaking is that significant either, I just did it as I spotted it as I know what radix primitives does from being loosely involved at the time, and it was easy to achieve the same thing here.

**Comment by devfixit** [comment]
> Could you tell more about your use-case? @devfixit
> 
> CSS tree shaking isn't really a thing, at least not in the same sense as relative to how JS tree shaking works. In theory something like https://purgecss.com could work, but we haven't tested it. Might be non-trivial to set it up.
> 
> We were thinking about exporting a CSS file that would cover just the layout components and their props (Box, Grid, Flex, Section, Container). Would that be helpful for what you are trying to achieve?

exactly that's what i need actually to have layout components css separate.
at the moment i'm getting entire css copied out when building(all the 25k lines of css.
@vladmoroz

**Comment by vladmoroz** [comment]
> I had been wondering about making a cli util that builds the radix themes css with postcss (with the radix themes internal postcss config) with certain components removed

This way you'd have to manually opt-into the components you need? In this case we can export all the components styles pre-built into the dist anyway, and then you'd be able to just import the CSS files for the components you plan to use.

**Comment by vladmoroz** [comment]
> exactly that's what i need actually to have layout components css separate.
> at the moment i'm getting entire css copied out when building(all the 25k lines of css.)

OK, we'll look into this, should be relatively easy. You might want to start using the 3.0 [release candidates](https://www.npmjs.com/package/@radix-ui/themes?activeTab=versions) early on as the next stable version will be based on that release.

**Comment by devfixit** [comment]
thanks very much for such a quick reply @vladmoroz @penx

**Comment by penx** [comment]
> This way you'd have to manually opt-into the components you need? In this case we can export all the components styles pre-built into the dist anyway, and then you'd be able to just import the CSS files for the components you plan to use.

good point, that would work well.

I was wondering if there are specificity issues that would mean the components need to be imported in a certain order to ensure weird bugs don't come up that aren't present in the standard build. I guess you do alphabetical order so as long as we keep to that it should be fine 🤞

**Comment by vladmoroz** [comment]
> I was wondering if there would be a chance of specificity issues that would mean the components need to be imported in a certain order to ensure weird bugs don't come up that aren't present in the standard build. I guess you do alphabetical order so as long as we keep to that it should be fine 🤞

That's definitely plausible, good callout

**Pull request merged**